### PR TITLE
 Support Rails 8 

### PR DESCRIPTION
**Pull request by ric2b**
This gem currently blocks updates to Rails 8.

I think it's best to just remove the upper version limit unless we know that there is an issue with future versions, otherwise every new Rails release requires a new release of the gem.

### CODE REVIEW

**Comment by MrAsler** [comment]
Rails 7.0 is only going to receive[ security updates ](https://rubyonrails.org/maintenance) until 25th of April of the next year, and testing rails 7.0 and rails 8.0 at the same time was being extremely problematic because Rails 7.0 needs Sqlite 1.x, whereas Rails 8.0 needs Sqlite 2.x and these library version checks are only done at runtime.

As such, the tests for Rails 7.0 were removed, although we still support this.

Additionally, the gem had some Rails version specific code that was no longer needed, so I removed it. In particular, our spec had the following code:
```
ActiveRecord::Base.belongs_to_required_by_default = true if ActiveRecord::VERSION::MAJOR >= 5
```
`belongs_to_required_by_default` is [true by default since Rails 5,](https://github.com/rails/rails/blob/eb7e42d5ff7cab2544fdbc64761dcca0e12752e7/railties/lib/rails/application/configuration.rb#L100) so we don't need this explicit configuration.

**Comment by erikkessler1** [comment]
We have an update to our guidelines that is relevant here I think: https://github.com/salsify/guidelines/pull/248

In particular, we have a number of services still on Ruby 3.1 and Rails 7.0 as those versions are still maintained, so I'm wary about removing test support for Ruby 3.1 and Rails 7.0.

**Comment by MrAsler** [comment]
@erikkessler1 This gem is very static in terms of changes,having 9 commits in the past 2 years and most of the commits were just bumping versions.

This means that if the consumer gems can use a previous version of this gem (0.20.0) if they need to update, and then after they update their rails version to 7.1, they can use this.

Regarding the Ruby version, let me see what I can do (if it is possible to set the minimum version to 3.1 and still test this with Rails 8)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 0.21.0
+- Allow use with future unknown Rails versions
+- Add support for Rails 8.0
+
```
Comment (ric2b) [inline CHANGELOG.md:6]: Would need to mention that Rails 7.0 support is also dropped (or untested)

**Comment by MrAsler** [comment]
@erikkessler1 @ric2b Ok so I added tests for Ruby 3.1.6, but only for Rails 7.1 and Rails 7.2.
We cannot test Rails 7.0 because of the problem with SQLite3 (namely, Rails 7.0 needs SQLite 1.x and Rails 8 needs SQLite 2.x and this check is only made at runtime, so the gemfile.lock generation fails).

My suggestion here is that for this repository we make an exception, where Rails 7.0 is untested, but for the new repos, we keep support and tests for Ruby 3.1 with Rails 7.0, 7.1 and 7.2 and Ruby 3.2+ tests with Rails 7.0, 7.1, 7.2 and 8.0.

File: CHANGELOG.md (hunk 1 above)
Comment (MrAsler) [reply CHANGELOG.md:6]: I'll make the necessary changes to the changelog after we decide on the way forward

**Comment by erikkessler1** [comment]
Can we get around the sqlite version conflict with a solution like this: https://github.com/salsify/goldiloader/pull/156?

**Comment by MrAsler** [comment]
@erikkessler1 That worked!

These are our tests:
```
- test:
    matrix:
      parameters:
        gemfile:
          - gemfiles/rails_7.0.gemfile
          - gemfiles/rails_7.1.gemfile
          - gemfiles/rails_7.2.gemfile
        ruby_version:
          - 3.1.6
          - 3.2.5
          - 3.3.5
- test:
    matrix:
      parameters:
        gemfile:
          - gemfiles/rails_8.0.gemfile
        ruby_version:
          - 3.2.5
          - 3.3.5

```

File: Appraisals [hunk 2]
```diff
@@ -1,8 +1,10 @@
 # frozen_string_literal: true
 
+
```
Comment (erikkessler1) [inline Appraisals:3]: Extra line.

```suggestion
```

File: rails_multitenant.gemspec [hunk 3]
```diff
@@ -26,8 +26,8 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 3.1'
 
-  spec.add_dependency 'activerecord', '>= 7.0', '< 7.3'
-  spec.add_dependency 'activesupport', '>= 7.0', '< 7.3'
+  spec.add_dependency 'activerecord', '>= 7.0'
+  spec.add_dependency 'activesupport', '>= 7.0'
```
Comment (erikkessler1) [inline rails_multitenant.gemspec:30]: As per our new guideline, we should have an upper bound.

```suggestion
  spec.add_dependency 'activerecord', '>= 7.0', '< 8.1'
  spec.add_dependency 'activesupport', '>= 7.0', '< 8.1'
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 0.21.0
+- Allow use with future unknown Rails versions
```
Comment (erikkessler1) [inline CHANGELOG.md:4]: Related to my comment about the upper bound on the version constraint in the gemspec, I don't think we want this.

**Comment by MrAsler** [comment]
@erikkessler1 I added the upper bounds and updated the changelog

**Review by erikkessler1 — APPROVED** [review]
🎉

**Pull request merged**